api/src/main/java/org/apache/iceberg/expressions/UnboundPredicate.java

```diff
@@ -67,29 +68,25 @@ Expression bind(Types.StructType struct) {
    * @throws ValidationException if literals do not match bound references, or if comparison on expression is invalid
    */
   public Expression bind(Types.StructType struct, boolean caseSensitive) {
-    Types.NestedField field;
-    String expressionFieldPath = ref().name();
 
-    boolean isNestedFieldExp = expressionFieldPath.indexOf('.') > -1;
-
-    field = isNestedFieldExp ? findNestedField(struct, expressionFieldPath, caseSensitive) :
-      caseSensitive ? struct.field(expressionFieldPath) : struct.caseInsensitiveField(ref().name());
+    Schema schema = new Schema(struct.fields());
+    Types.NestedField field = schema.findField(caseSensitive? ref().name(): ref().name().toLowerCase());
```

Comment:
This should probably use a case insensitive version of `findField` instead of assuming that passing lower-case in will work.

Reply:
will do.